File: web/src/api/merchant.ts
// web/src/api/merchant.ts
import request from './request'
import type { ApiResponse, PageData, Merchant, Order, Withdrawal, MerchantBalanceLog, Refund } from './types'

 const merchantApiBase = '/api/merchant'

// 注册
export function register(data: { username: string; password: string; email?: string; phone?: string }) {
  return request.post<any, ApiResponse>(`${merchantApiBase}/auth/register`, data)
}

// 登录
export function login(data: { username: string; password: string }) {
  return request.post<any, ApiResponse>(`${merchantApiBase}/auth/login`, data)
}

// 仪表盘
export function getDashboard() {
  return request.get<any, ApiResponse>(`${merchantApiBase}/dashboard`)
}

// 获取个人信息
export function getProfile() {
  return request.get<any, ApiResponse<Merchant>>(`${merchantApiBase}/profile`)
}

// 更新个人信息
export function updateProfile(data: { email?: string; phone?: string }) {
  return request.put<any, ApiResponse>(`${merchantApiBase}/profile`, data)
}

// 修改密码
export function updatePassword(data: { old_password: string; new_password: string }) {
  return request.put<any, ApiResponse>(`${merchantApiBase}/profile/password`, data)
}

// 重置API密钥
export function resetApiKey() {
  return request.post<any, ApiResponse<{ api_key: string }>>(`${merchantApiBase}/profile/reset-key`)
}

// 订单列表
export function getOrders(params: { page: number; page_size: number; status?: number }) {
  return request.get<any, ApiResponse<PageData<Order>>>(`${merchantApiBase}/orders`, { params })
}

// 订单详情
export function getOrder(tradeNo: string) {
  return request.get<any, ApiResponse<Order>>(`${merchantApiBase}/orders/${tradeNo}`)
}

// 提现列表
export function getWithdrawals(params: { page: number; page_size: number; status?: number }) {
  return request.get<any, ApiResponse<PageData<Withdrawal>>>(`${merchantApiBase}/withdrawals`, { params })
}

// 申请提现
export function applyWithdrawal(data: {
  amount: string
  account_type: string
  account_no: string
  account_name: string
}) {
  return request.post<any, ApiResponse>(`${merchantApiBase}/withdrawals`, data)
}

// 余额日志
export function getBalanceLogs(params: { page: number; page_size: number }) {
  return request.get<any, ApiResponse<PageData<MerchantBalanceLog>>>(`${merchantApiBase}/balance-logs`, { params })
}

// 退款管理
export const createRefund = (data: {
  trade_no: string
  amount: string
  reason?: string
  notify_url?: string
}) => request.post<any, ApiResponse>(`${merchantApiBase}/refunds`, data)

export const getRefunds = (params: any) =>
  request.get<any, ApiResponse<PageData<Refund>>>(`${merchantApiBase}/refunds`, { params })

// 商户测试支付
export function testPayment(data: { amount: string; pay_type: string; pay_method?: string }) {
  return request.post<any, ApiResponse>(`${merchantApiBase}/test-payment`, data)
}

File: internal/router/merchant.go
package router

import (
	"github.com/example/epay-go/internal/api/merchant"
	"github.com/example/epay-go/internal/middleware"
	"github.com/example/epay-go/pkg/jwt"
	"github.com/gin-gonic/gin"
)

func registerMerchantRoutes(r *gin.Engine) {
	merchantAPI := r.Group("/api/merchant")
	{
		merchantAPI.POST("/auth/login", merchant.Login)
		merchantAPI.POST("/auth/register", merchant.Register)

		merchantAuth := merchantAPI.Group("")
		merchantAuth.Use(middleware.JWTAuth(jwt.TokenTypeMerchant))
		{
			merchantAuth.POST("/auth/logout", merchant.Logout)
			merchantAuth.GET("/dashboard", merchant.Dashboard)

			merchantAuth.GET("/profile", merchant.GetProfile)
			merchantAuth.PUT("/profile", merchant.UpdateProfile)
			merchantAuth.PUT("/profile/password", merchant.UpdatePassword)
			merchantAuth.POST("/profile/reset-key", merchant.ResetAPIKey)

			merchantAuth.GET("/orders", merchant.ListOrders)
			merchantAuth.GET("/orders/:trade_no", merchant.GetOrder)

			merchantAuth.GET("/settlements", merchant.ListSettlements)
			merchantAuth.POST("/settlements", merchant.ApplySettlement)

			merchantAuth.GET("/records", merchant.ListRecords)

			merchantAuth.POST("/refunds", merchant.CreateRefund)
			merchantAuth.GET("/refunds", merchant.ListRefunds)

			merchantAuth.POST("/test-payment", merchant.TestPayment)
		}
	}
}
